### ISSUE TITLE
Error: Filter cant be parsed in get-csonlineuser

### ISSUE THREAD

**Issue by eloka080677** [MicrosoftDocs/office-docs-powershell#8919]
Hi There, just have problem with get-csonlineuser in combination with the filter parameter.
As described on the page you can filter the output of get-csonlineuser like 

`get-csonlineuser  -filter {usagelocation -eq "DE"}`

but with the latest release 3.1.1 and 3.1.0 i get always an error message, that the filter cant be parsed

`get-csonlineuser  -filter {usagelocation -eq "DE"}
 Search-CsUser_Search: C:\Program Files\WindowsPowerShell\Modules\MicrosoftTeams\3.1.1\netcoreapp3.1\custom\PsExt\Get-CsUserSearch.ps1:52
Line |
  52 |              Microsoft.Teams.ConfigAPI.Cmdlets.internal\Search-CsUser  …
     |              ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
     | Filter couldn't be parsed
`

there is no information that filter syntax has changed.

using OPATH Filters syntax, is working

`get-csonlineuser  -filter "usagelocation eq 'DE'" | select userprincipalname
UserPrincipalName
svc_RequestATeam@galdev.onmicrosoft.com
Tim.Testuser@galdev.onmicrosoft.com`

is this a planned change or is there a bug in the new modules ?

thanks
---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: 7008ec10-a056-d8bb-bb4e-b4690a84adfd
* Version Independent ID: 2dcb78d9-7f36-208e-0989-184905d73011
* Content: [Get-CsOnlineUser (SkypeForBusiness)](https://docs.microsoft.com/en-us/powershell/module/skype/get-csonlineuser?view=skype-ps)
* Content Source: [skype/skype-ps/skype/Get-CsOnlineUser.md](https://github.com/MicrosoftDocs/office-docs-powershell/blob/master/skype/skype-ps/skype/Get-CsOnlineUser.md)
* Service: **skypeforbusiness-powershell**
* GitHub Login: @tomkau
* Microsoft Alias: **tomkau**

**Comment by dariomws** [comment]
Hi @eloka080677, thank you for your feedback.

@isrumnon, could we provide more information? Thanks!

**Comment by isrumnon** [comment]
Hi @eloka080677  @dariomws , this is a bug in our system and we are targeting to support the old format soon. We are targeting end of March to reintroduce the old formats. And you are correct in pointing out that presently the OPATH syntax is working - you can continue to follow this syntax or we can deflight your tenant from our recent changes if that helps. Do let us know if you want to go with the second option. Thanks

**Comment by DEberhardt** [comment]
> We are targeting end of March to reintroduce the old formats

Er. This is six weeks in which all our deployments are hampered on 10+ tenants. This is going to impact a lot of our customers. I am afraid this will go down swimmingly with our operations team...

Is there an option to downgrade the MicrosoftTeams version (3.0, 2.6?) to circumvent this?

**Comment by isrumnon** [comment]
Hi @DEberhardt we have a workaround to re-enable the old syntax for your affected tenants. Can you share the tenant IDs of the tenants impacted by this change so we can unblock you?

**Comment by DEberhardt** [comment]
I can collect them tomorrow morning (already signed off for today). Should this be resolved via a ticket to be opened via the tenant? We need to play this visibly for each customer

**Comment by DEberhardt** [comment]
Are you aware that this also breaks `Grant-CsOnlineVoiceRoutingPolicy` assignments as the CmdLet cannot search for users (Error: UserId not found) - here too. Don't know if this is one bug or two, but it cripples MACD requests on our end for DirectRouting customers

**Comment by isrumnon** [comment]
To your first question: We have already rolled the change to most affected tenants but I want to confirm by looking at that list you will be sharing. And you can just share the tenant IDs with me (either through Teams or e-mail).

To your second question: No my team does not own that cmdlet. Would suggest you reach out to pbafna@microsoft.com

**Comment by eloka080677** [comment]
Sorry for being late with reply.
This problem occures in different tenants, so it is hard t o write down all customer tenants where we are supporting implementing Microsoft Teams. Glad to hear you are rolling back the changes. I will try today if the problem is solved. If i got still problems i will post the tenant id

**Comment by isrumnon** [comment]
Sure, Thanks!

**Comment by yogkumgit** [comment]
@dariomws and @isrumnon Thank you very much for the contribution and sharing this explanation. @eloka080677 Hope this comment is helpful for you. If you see a documentation update is required, please feel free to open an issue for the same. We proceed here to close it. Thanks for taking out some time to open the issue. Appreciate and encourage you to do the same in future also

**Comment by man4ini** [comment]
Hi, used to work with OPATH 1 previous week, not anymore. 

![image](https://user-images.githubusercontent.com/99261750/154468366-40ff0185-cd6f-472f-bc54-109f66864914.png)

**Comment by isrumnon** [comment]
Hi @man4ini we had to rollback that change as most customers were not comfortable with the OPATH syntax. Can you please revert to the old syntax?

**Comment by man4ini** [comment]
![image](https://user-images.githubusercontent.com/99261750/154476430-34aca174-c624-41fe-86e7-5cc2015928b9.png)

**Comment by man4ini** [comment]
![image](https://user-images.githubusercontent.com/99261750/154477557-2ae2454a-6b74-49b3-a68f-3db3ec891b4f.png)

**Comment by isrumnon** [comment]
@man4ini I just noticed, the error message you are getting in the first message is because non-numeric characters are not supported and you can still use the OPATH syntax
![image](https://user-images.githubusercontent.com/89481047/154481296-33f3035f-f501-4086-a69a-5f42913fc2fc.png)

**Comment by man4ini** [comment]
@isrumnon , as said I can still use it but as you can see it does not provide an output. It should :) We have this highly implemented in our automations.

**Comment by DEberhardt** [comment]
Hello,

I generally use the filter like this:

```powershell
# Equals
$Filter = 'VoicePolicy -eq "BusinessVoice"'

# Like (Number)
$PhoneNumber = "123456789"
$Filter = 'LineURI -like "*{0}*"' -f "$PhoneNumber"

# Like (TEL URI)
$PhoneNumber = "tel:+123456789"
$Filter = 'LineURI -like "*{0}*"' -f "$PhoneNumber"

# Call
$CsUser = Get-CsOnlineUser -Filter $Filter
```

If I recall correctly OPATH (3.0?) did not have a LIKE query which stumped us for queries against AzureAd a bit. Has this since been addressed?

**Comment by DEberhardt** [comment]
Making this a separate post - it fits thematically here, but may need a separate issue altogehter:

The LineUri now is no longer presented as a full TEL URI string ("tel:+" is omitted _sometimes?_), but input is accepted. This impacts on queries that go for `$Filter = 'LineURI -like "tel:+*"'` but I noticed that the output is still returned... is this a formatting issue on the output side of `Get-CsOnlineUser` and when queried, the object still has the full TEL URI string present? If so, could this be looked at to provide the consistency we'd expect from this parameter please?

Formats encountered:
```powershell
123456789012345 # For CallingPlan Numbers as they are recorded as is from `Get-CsOnlineTelephoneNumer`
123456789012345 # For (some) DirectRouting Numbers
123456789012345;ext=1234 # For (some) DirectRouting Numbers with extensions
tel:+1234567890 # For (some) DirectRouting Numbers
tel:+1234567890;ext=12345 # For (some) DirectRouting Numbers with extensions
```

I would anticipate (or: welcome) that all Numbers are displayed the same way (preferably with a full Tel URI) - i.E. line 4 or 5 above. If the desire is to move to a new presentation, I understand that "tel:+" may be seen as superflous, but it has had very neat advantages when identifying this string.
Specifically, and that is a use case I have every day, when exporting output from `Get-CsOnlineUser` as CSV and managing it in Excel... mainly because Excel likes to interpret strings that are numbers as such, and if they are longer than 10 digits (like literally every phone number out there) it presents it in scientific form, `1.23 E08` which is great but unusable for phone numbers. Keeping the LineUri presented as a Tel Uri will tell Excel that this must be a string as there are non-numeric digits present.
Also, Excel likes to interpret a + as a mathematical operator and automatically strips it away, which leads to fun and games when opening and saving CSV files containing numbers in E.164 format in Excel.

**Comment by man4ini** [comment]
Thanks God everything is fine in 2.6.0 :)

DEberhardt, tried your approach in 2.6.0 and cannot get an output. 

![image](https://user-images.githubusercontent.com/99261750/154496778-0f96668c-fc22-4def-b526-386afd7adf10.png)

**Comment by man4ini** [comment]
My bad :) run the variable and got it

**Comment by man4ini** [comment]
Hey folks, when we saw this we started to prepare but looks like there is a long path to go:

![image](https://user-images.githubusercontent.com/99261750/154502984-cbbfdc4f-b7d0-4b12-8368-9afa7dcc152d.png)

The new commandlets are not available in 2.6.0 which is working fine! If you retire the old commandlets it means that when running the new commandlets you will get the newest Teams module and IF all of this stuff is not fixed it will create a huge mess. We are mid of February now so please think twice before acting.

**Comment by isrumnon** [comment]
@DEberhardt thanks for bringing this to our notice - we'll look into it

**Comment by man4ini** [comment]
@isrumnon , sorry to tell you but you have no idea what is going on. I see that you are a project manager and maybe that explains it. 

I had to play around a lot to figure out all of this. We have 3.1.1 on our dev instance and found out that it is not working as expected.

In Prod we have 2.6.0 which is stable and everything is fine there.

**Comment by vincentkurmi** [comment]
Hi @man4ini, I found some points to improve here. https://github.com/MicrosoftDocs/office-docs-powershell/issues/8973

It can may be help you a little bit.

**Comment by isrumnon** [comment]
@man4ini apologies, there are some nuances here which makes it hard to pin-point the problem you might be facing. I mean that in the sense that there are certain customers that are still enabled for OPATH syntax while the rest are not. 
If you can share the tenant ID in which you are facing problems for TPM 3.1.1 we can help get you unblocked to use it without facing issues for the filtering scenarios

**Comment by isrumnon** [comment]
@DEberhardt also can you share the tenant ID in which you were facing problems with the LIKE query in TPM 3.0? We might be able to fix that issue

**Comment by man4ini** [comment]
@isrumnon, we are human beings at the end and we are not perfect, I do understand that but last week I was able to use OPATH syntax and it worked, now it is not working no matter I do. Tried with the other approach as well:

![image](https://user-images.githubusercontent.com/99261750/154511920-408109c9-9f63-481c-8878-2d14244142a0.png)

Few hours ago you told me:

![image](https://user-images.githubusercontent.com/99261750/154512140-e3e63d2e-952d-442a-97ee-55b9f720126c.png)

and I tried the old syntax and it is still not working. What kind of unblocking you are referring to? I can still use 2.6.0 to do my job and would like to stick to 3.1.1 in our test environment so I can track progress. Really need to know what kind of unblocking you are talking about because it might be just another variable I need to worry in the equation.

**Comment by isrumnon** [comment]
@man4ini I had earlier assumed that your tenant was part of the batch of tenants that were enabled for the old syntax since you mentioned that OPATH syntax was not working. But it seems like even the OPATH syntax is not working - for this reason I am requesting you to share the tenant ID on which you are using TPM 3.1.1 so we can revert it back to be able to use the old syntax and your other scenarios should work as expected. Hope this helps

**Comment by man4ini** [comment]
@isrumnon, let's give it a try a8029e02-891d-4bb6-86d1-1831ab11ed72. Thanks!

**Comment by isrumnon** [comment]
@man4ini can you please check now?

**Comment by man4ini** [comment]
@isrumnon, confirm it is working now. Are you going to change the syntax going forward for 3.1.1?

**Comment by lgaudreau** [comment]
Is this post still valid to comment in?  My tenant uses the 'tel:+1xxxxxxxxxx;ext=xxxx' format and I have no way to filter by number using the 4.1.0 module.  
```
$Number = "tel:+1xxxxxxxxx;ext=xxxx"
$filter = 'LineURI -eq "{0}"' -f $Number
Get-CsOnlineUser -Filter $filter
Get-CsOnlineUser: Phone number has non digit chars. tel:+1xxxxxxxxx;ext=xxxx
```
This is the same result if I try running the original style filter:

```
Get-CsOnlineUser -Filter 'LineURI -eq "tel:+1xxxxxxxxx;ext=xxxx"'
Get-CsOnlineUser: Phone number has non digit chars. tel:+1xxxxxxxxx;ext=xxxx
```

Is there another format I can try?

**Comment by coyvelano** [comment]
Hi @lgaudreau 

I'm having the same issue on my end - shall we try 2.6.0?

**Comment by lgaudreau** [comment]
@coyvelano, 2.6.0 works fine for the filter, but is going to be deprecated.  Not sure what the solution is when that happens if the filtering isn't fixed in 4.1.0 or the next module.

**Comment by BalajiNJL** [comment]
I want to re-open this bug. Today I installed MicrosoftTeam PS module v 4.4.1 and I am having exactly the same issue. I am not able to run -Filter command. Follow command fails with error below

**Get-CsOnlineUser -Filter {Enabled -eq $True -and EnterpriseVoiceEnabled -eq $true}** 

_[DBG]: PS C:\Users\Administrator> Get-CsOnlineUser -Filter {Enabled -eq $True -and EnterpriseVoiceEnabled -eq $true}

Get-CsOnlineUser: C:\Users\Administrator\Documents\PowerShell\Modules\MicrosoftTeams\4.4.1\exports\ProxyCmdletDefinitions.ps1:11344:23
 Line |
11344 |          $scriptCmd = {& $wrappedCmd @PSBoundParameters}
      |                        ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
      | Query not supported for property: Enabled query: Enabled -eq $True -and EnterpriseVoiceEnabled -eq $true position: 1. Parameter name: $filter

[DBG]: PS C:\Users\Administrator>_

I also tried 
Get-CsOnlineUser | Where {$_.Enabled -eq $True -and $_.EnterpriseVoiceEnabled -eq $true} 

and it is not returning any records.

What is the fix? Do I need to rollback to previous version? I can reproduce this issue with tenant id: : d5df4d89-c307-4bda-a2a9-e01eb26b48ff

**Comment by coyvelano** [comment]
@BalajiNJL  I rolled back to 2.6.0

**Comment by isrumnon** [comment]
Hi @BalajiNJL @coyvelano "Enabled" has been renamed to "IsSipEnabled", query should work once you've made this change
Reference - https://docs.microsoft.com/en-us/powershell/module/skype/get-csonlineuser?view=skype-ps

@lgaudreau your original query should be working now in latest TPM modules

**Comment by BalajiNJL** [comment]
Thanks @isrumnon - changing to IsSipEnabled worked.
